pkg/controller/sync/accessor.go

```diff
@@ -240,6 +236,13 @@ func (a *resourceAccessor) FederatedResource(eventSource util.QualifiedName) (Fe
 
 	var namespace *unstructured.Unstructured
 	if a.targetIsNamespace {
+		if federatedName.Namespace != federatedName.Name {
+			// A FederatedNamespace is only valid for propagation
+			// if it has the same name as the containing namespace.
+			a.eventRecorder.Eventf(
+				resource, corev1.EventTypeWarning, "InConsistentNameNamespace", "Name and namespace are not consistent")
```

Comment:
`s/InConistentNameNamespace/InvalidName/`

`s/Name and namespace are not consistent/The name of a federated namespace must match the name of its containing namespace./`

Reply:
@font Maybe federated resource validation, in addition to dry-run, could have special handling for federated namespaces to prevent creation when the name doesn't match that of the containing namespace?

Reply:
done, thanks for the comment.